CMSgov/mint-app
# Easi 2063/it tools fe

# EASI-2063

## Changes and Description

- Completed IT Tools Task List Section
- Create a custom component `ITToolsSummary` for rendering related task list questions, answers, and subtext
- Created a custom Form Component for capturing followup IT Tools answers
- Created a Cypress command for cycling through IT Tools testing flow

![Screen Shot 2022-07-21 at 10 20 27 AM](https://user-images.githubusercontent.com/95709965/180237397-6b4ecb48-2c88-4bcc-a543-682da9eaf712.png)

[Figma](https://www.figma.com/file/M72DReE73SSXMN48WZfaKN/MINT---New-Model-Process-(Model-Intake)?node-id=2156%3A183370)

[Figma - logic/flow](https://www.figma.com/file/M72DReE73SSXMN48WZfaKN/MINT---New-Model-Process-(Model-Intake)?node-id=2156%3A183370)

## Notes

- Cypress testing has been flaky, and may not pass every time.  This flakiness will be covered in a future ticket.

- Some Payments IT Tools/Cypress integrations are not complete, as they are dependent upon Payments task list to be complete.  May hold this PR open until Payments is merged.

## How to test this change

- Checkout branch
- Either seed an existing plan or create a new one. 
- Navigate to the IT Tools task list section
- For each question, click "Go to the question" to either answer or change and answer. 
- Navigate back to the IT Tools page (can either wait for autosave and click browser's back button, or save and go to the task list and back to the original IT Tools page)
- Reference the Figma's logic/flow link to ensure the correct pattern of requirements

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
